web/src/engine/websites/CoolMic.ts

```diff
@@ -0,0 +1,81 @@
+import { Tags } from '../Tags';
+import icon from './CoolMic.webp';
+import { Chapter, DecoratableMangaScraper, Page, type Manga } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { Fetch, FetchCSS, FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+import { Exception } from '../Error';
+import { WebsiteResourceKey as R } from '../../i18n/ILocale';
+
+type JsonChapters = {
+    episodes: {
+        id: number,
+        number : string
+    }[]
+}
+
+type APIPages = {
+    image_data?: {
+        path: string
+    }[],
+    signed_cookie?: CookieSigner
+}
+
+type CookieSigner = {
+    'CloudFront-Policy': string,
+    'CloudFront-Signature': string,
+    'CloudFront-Key-Pair-Id': string
+}
+
+@Common.MangaCSS(/^{origin}\/titles\/\d+$/, 'meta[property="og:title"]')
+@Common.MangasNotSupported()
+
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/v1/`;
+
+    public constructor() {
+        super('coolmic', 'CoolMic', 'https://coolmic.me', Tags.Media.Manhwa, Tags.Media.Manga, Tags.Language.English, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const [jsonNode] = await FetchCSS(new Request(new URL(manga.Identifier, this.URI)), 'title-page');
```

Comment:
May use a more appropriate CSS query parameter as `title-page` is likely a custom tag which is a bit confusing